src/mds/MDCache.cc

```diff
@@ -6808,6 +6915,11 @@ void MDCache::handle_cache_expire(MCacheExpire *m)
 	      dir = *q;
 	      if (dir->is_replica(from)) {
 		dout(7) << " dir expire on " << *dir << " from mds." << from << dendl;
+                dout(20) << " dir replicated=" << dir->get_num_ref(MDSCacheObject::PIN_REPLICATED) << " replica_map size=" << dir->replica_map.size() << dendl;
+                for (std::map<mds_rank_t,unsigned>::iterator i=dir->replica_map.begin(); i != dir->replica_map.end(); ++i) {
+                  dout(20) << "  rep " << i->first << ":" << i->second << dendl;
+                }
+                
```

Comment:
> ostream& operator<<(ostream& out, const CDir& dir) already contains code to print replica_map, why adding this?

This was some spurious debug from another issue, I've removed it now.

> The rest changes look good

Cool.  I'm going to merge this once I've tested it again.